shuffle/cli/src/node.rs

```diff
@@ -1,27 +1,39 @@
 // Copyright (c) The Diem Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::new::Config;
 use anyhow::Result;
 use diem_config::config::NodeConfig;
+use diem_genesis_tool::validator_builder::ValidatorConfig;
+use diem_types::{on_chain_config::VMPublishingOption};
+use dir::home_dir;
 use std::{fs, path::Path};
 
-pub fn handle(project_path: &Path) -> Result<()> {
+pub fn handle() -> Result<()> {
     // TODO: Generate prefunded accounts
-    let node_config = NodeConfig::load(project_path.join("nodeconfig/0/node.yaml").as_path())?;
-    let config = read_config(project_path)?;
+    let node_config_path = &(home_dir().unwrap().join(".shuffle"));
+    if !Path::new(node_config_path).is_dir() {
+        println!("Creating node config in {}", node_config_path.display());
+        fs::create_dir_all(node_config_path)?;
+        generate_validator_config(node_config_path);
+    } else {
+        println!("Accessing node config in {}", node_config_path.display());
+    }
+    let node_config = NodeConfig::load(node_config_path.join("nodeconfig/0/node.yaml").as_path())?;
     println!(
-        "running shuffle node configured for {} in {}",
-        &config.blockchain,
-        project_path.display()
+        "running shuffle node in {}",
+        node_config_path.display()
     );
     diem_node::start(&node_config, None);
     Ok(())
 }
 
-fn read_config(project_path: &Path) -> Result<Config> {
-    let config_string =
-        fs::read_to_string(project_path.join("Shuffle").with_extension("toml")).unwrap();
-    let read_config: Config = toml::from_str(config_string.as_str())?;
-    Ok(read_config)
+fn generate_validator_config(node_config_path: &Path) -> Result<ValidatorConfig> {
+    let publishing_option = VMPublishingOption::open();
+    // TODO (dimroc): place genesis module deployment/generate validator config
+    // in shuffle node command
+    shuffle_custom_node::generate_validator_config(
```

Comment:
i'm confused where is `shuffle_custom_node` instantiated?

Reply:
Hmm yea that's weird. I took it from https://github.com/diem/diem/blob/main/shuffle/cli/src/new.rs where shuffle_custom_node isn't instantiated either.

Reply:
It's a crate pulled in via `Cargo.toml`. This name was made before us by Emma so let's just keep it as is for now.

https://github.com/diem/diem/blob/9a04ed9fa8695f1e6b1175af97d4d10e0fc6ed1b/shuffle/cli/Cargo.toml#L44